JSONAdapterInterface/IJSONValue.h

```diff
@@ -80,6 +90,29 @@ namespace systelab { namespace json {
 		// Factory methods
 		virtual std::unique_ptr<IJSONValue> buildValue(Type) const = 0;
 		virtual std::unique_ptr<IJSONDocument> buildDocument() const = 0;
+
+	private:
+
+		// Array iterator for range-loop
+		template <typename BaseIterator_, typename Ref_>
+		class ArrayIterator_ : public BaseIterator_
```

Comment:
The implementation of the whole class has changed and is not related to the vector<unique_ptr<IJSONValue>> (because it reveals the current way to store values contained in the JSONValue implementation, something inherent in the implementation but not in the interface)